etcd/etcd.go

```diff
@@ -204,13 +203,13 @@ func (c *Client) tryEndpoints(mreq func(string) (*http.Request, error)) (*http.R
 		}
 
 		endpointErrs = append(endpointErrs, err)
-		endpoints = endpoints[1:]
 	}
 
-	rotate := len(c.endpoints) - len(endpoints)
-	c.endpoints = append(c.endpoints[rotate:], c.endpoints[:rotate]...)
+	if lastEndpoint > 0 {
```

Comment:
this would consider it an error if a single endpoint fails, while the original considered the outcome an error only if all the endpoints failed.